streams/src/test/java/org/apache/kafka/streams/kstream/JoinWindowsTest.java

```diff
@@ -101,102 +93,100 @@ public void startTimeShouldNotBeAfterEnd() {
         }
     }
 
-    @Test
-    public void untilShouldSetGraceDuration() {
```

Comment:
We should avoid outright removing tests that cover the old deprecated APIs until we actually get to remove those methods. Instead we can just scope the deprecation warning suppression to the individual tests that cover the behavior of the old APIs like here, and a few places below